rcgen/tests/webpki.rs

```diff
@@ -491,15 +491,38 @@ fn test_certificate_from_csr() {
 	params
 		.distinguished_name
 		.push(DnType::CommonName, "Dev domain");
+
 	let cert_key = KeyPair::generate().unwrap();
 	let csr = params.serialize_request(&cert_key).unwrap();
 	let csr = CertificateSigningRequestParams::from_der(csr.der()).unwrap();
 
 	let (mut params, ca_key) = util::default_params();
 	params.is_ca = IsCa::Ca(BasicConstraints::Unconstrained);
+	let eku_test = vec![
```

Comment:
I'm a bit confused by this test. We put EKUs into parameters, check the expected EKUs are in the parameters, create a self-signed CA certificate from the parameters, and then verify the CA certificate's parameters contain the EKUs.

It doesn't seem like anything is testing that a CSR that has EKUs is loaded with those EKUs, or that a CSR created from params with EKUs preserve the EKUs. I think those are the important properties we want to test right?

Reply:
I was a bit stumped as how to fully write a test here since some of the testing I had thought of required for  CertificateParams::serialize_request to implement the extended key usage, but thought adding that might be better for a different pull request.

Reply:
> adding that might be better for a different pull request.

I would be OK with it being part of this PR!